src/analysis/sign-lattice.cpp

```diff
@@ -0,0 +1,60 @@
+#include <type_traits>
+#include "lattice.h"
+
+namespace wasm::analysis {
+
+struct SignLattice {
+public:
+    enum Sign {
+        BOTTOM,
+        NEGATIVE,
+        ZERO,
+        POSITIVE,
+        TOP
```

Comment:
Even if we use "any" and "none" for the elements of this particular lattice, we would still want to use `getTop` and `getBottom()` for the lattice methods because top and bottom do not always correspond to elements that could be referred to as "any" or "none". Reading this analysis code is going to require some familiarity with the relevant formalisms, so I think we should lean into using terms from the literature and have ample documentation for readers and users to bridge the gap.